nutrition_labels/grant_tagger.py

```diff
@@ -216,20 +179,59 @@ def split_data(self, X_vect, y):
             ]
             # Make sure they are in order otherwise it'll be all 1111s and then 0000s
             sample_index.sort()
-            y = [y[i] for i in sample_index]
-            if self.vectorizer_type == "bert":
-                X_vect = [X_vect[i] for i in sample_index]
-            else:
-                X_vect = X_vect[sample_index]
+
+            # Store data not in sample
+            not_sample_index = data.index.difference(sample_index)
+            unused_data = data.iloc[not_sample_index, :]
+            self.X_unused = unused_data['Grant texts'].tolist()
+            self.y_unused = unused_data[self.label_name].tolist()
+            self.unused_ids = unused_data[train_data_id].tolist()
+            # Sample data
+            data = data.iloc[sample_index, :]
+        else:
+            self.X_unused = []
+            self.y_unused = []
+            self.unused_ids = []
+
+        X = data[["Grant texts", train_data_id]]
+        y = data[self.label_name].tolist()
 
         # Randomly shuffled at the beginning so turn this off
         X_train, X_test, y_train, y_test = train_test_split(
-            X_vect, y, test_size=self.test_size, shuffle=False
-        )
+                X, y, test_size=self.test_size, shuffle=False
+            )
+
+        self.X_train_ids = X_train[train_data_id].tolist()
```

Comment:
why do you save that? note that if you rely on the user keeping the data in the same order so that ids maps to training examples, I think this can lead to bugs. a user can receive X_train and change the order by shuffling.

Reply:
I save it to be able to link IDs to whether they were in the test/train/unseen split, since this was something I wanted for my pipeline.

If you're pipeline is (as mine is in `train_several_models`):
```
X_train, X_test, y_train, y_test = grant_tagger.split_data(training_data, train_data_id)
grant_tagger.fit(X_train, y_train)
grant_info = grant_tagger.train_test_info(X_test, y_train, y_test)
```
then you will save out a nice file of `IDs - actual Y - predicted Y - whether this ID is in the train/test/unseen split`.

You are right that if you did
```
X_train, X_test, y_train, y_test = grant_tagger.split_data(training_data, train_data_id)
grant_tagger.fit(X_train, y_train)
X_test = X_test[shuffle]
y_train = y_train[shuffle]
y_test = y_test[shuffle]
grant_info = grant_tagger.train_test_info(X_test, y_train, y_test)
```
you could mess up this output file (if indeed the user even wanted to output this file).